kivy/core/text/markup.py

```diff
@@ -752,7 +752,14 @@ def p_restricted(line, uw, c):
         s = self.get_extents(last_text)
         if len(last_text):
             line1.append(LayoutWord(last_word.options, s[0], s[1], last_text))
-        elps.options = last_word.options
+
+        elps_opts = old_opts
```

Comment:
What was the change, and how did it not fix it? As for `textwidth`, check the function from the providers and you'll see it relies on the current options. That's why self.option need to be updated and restored before doing that.